frontendservice/src/views/CreateShop.vue

```diff
@@ -0,0 +1,160 @@
+              <option>Furniture</option>
+              <option>Electronics</option>
+              <option>Jewellery</option>
+              <option>Agriculture</option>
+            </select>
+          </div>
+        </div>
+      </div>
+      <div class="field">
+        <div class="control">
+          <label class="checkbox">
+            <input type="checkbox" id="conditions-check" />
+            I agree to the
+            <a href="#">terms and conditions</a>
+          </label>
+        </div>
+        <div v-if="hidden" id="msg"></div>
+      </div>
+      <div class="field is-grouped">
+        <div class="control">
+          <button class="button is-link" type="submit" value="submit">Submit</button>
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import shopOwnerNavbar from "../components/NavBar-ShopOwner";
+import Vue from "vue";
+import VueTelInput from "vue-tel-input";
+import firebase from "firebase";
+
+Vue.use(VueTelInput);
+
+export default {
+  name: "CreateShop",
+  components: {
+    shopOwnerNavbar
+  },
+  data() {
+    return {
+      hidden: true,
+      isCorrect: false,
+      phoneNumber: 0
+    };
+  },
+  methods: {
+    createShop: function() {
+      var shopName = document.getElementById("shopName").value;
+      var address = document.getElementById("address").value;
+      var shopType = document.getElementById("shopType").value;
+      var checkBox = document.getElementById("conditions-check");
+      var phoneNo = this.phoneNumber;
+      console.log(document.getElementById("phoneNumber"));
+      // Regex check only permit # . characters, digits and space
+      // eg - Haridwar123 is invalid.
+      var addressRegex = new RegExp(/^[#.a-zA-Z0-9 ]+$/i);
+      // Regex check only permit characters, digits and space
+      // eg - Ishita Sweets is invalid.
+      var nameRegex = new RegExp(/^[a-zA-Z0-9 ]+$/i);
+
+      if (!checkBox.checked) {
+        var checkBoxError = document.createElement("div");
+        checkBoxError.innerHTML = "Please check this to continue.";
+        document.getElementById("msg").appendChild(checkBoxError);
+      }
+      if (!addressRegex.test(address)) {
+        var addressError = document.createElement("div");
+        addressError.innerHTML = "Invalid address.";
+        document.getElementById("msg").appendChild(addressError);
+      }
+      if (!nameRegex.test(shopName)) {
+        var nameError = document.createElement("div");
+        nameError.innerHTML = "Invalid Shop Name.";
+        document.getElementById("msg").appendChild(nameError);
+      }
+      if (!this.isCorrect) {
+        var phoneNumberError = document.createElement("div");
+        phoneNumberError.innerHTML = "Invalid Phone Number.";
+        document.getElementById("msg").appendChild(phoneNumberError);
+      }
+      if (checkBox.checked && addressRegex.test(address) && nameRegex.test(shopName) && this.isCorrect) {
```

Comment:
Can we just return from the other error "if" conditions and not have this condition here?

Reply:
yeah did it!